test/content/support.js

```diff
@@ -202,39 +202,25 @@ function resetDB() {
  * Equivalent to JSON.stringify, except that object properties are stringified
  * in a sorted order.
  */
-function stableStringify(obj, level, label) {
-	if (!level) level = 0;
-	let indent = '\t'.repeat(level);
-	
-	if (label) label = JSON.stringify('' + label) + ': ';
-	else label = '';
-	
-	if (typeof obj == 'function' || obj === undefined) return null;
-	
-	if (typeof obj != 'object' || obj === null) return indent + label + JSON.stringify(obj);
-	
-	if (Array.isArray(obj)) {
-		let str = indent + label + '[';
-		for (let i=0; i<obj.length; i++) {
-			let json = stableStringify(obj[i], level + 1);
-			if (json === null) json = indent + '\tnull'; // function
-			str += '\n' + json + (i < obj.length-1 ? ',' : '');
+function stableStringify(obj) {
+	function recompose (obj) {
+		var keys = Object.keys(obj);
+		keys.sort();
+		var o = {};
+		for (let key in obj) {
+			o[key] = obj[key];
+			delete obj[key];
+		}
```

Comment:
You could skip this loop and fold everything into one. More importantly though, we should not alter the original object, but create a new one for the purposes of serialization, because we would destroy any enumerable and configurable getters/setters defined on the object.
```javascript
let o = {},
  keys = Object.keys(obj).sort();
for (let i=0; i<keys.length; i++) {
  o[keys[i]] = obj[keys[i]];
}
return o;
```